src/App.tsx

```diff
@@ -1,27 +1,95 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import 'bulma/css/bulma.min.css';
+import { Vortex } from 'react-loader-spinner';
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import { GoodsList } from './components/GoodList';
+import { Button } from './components/Button';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [clickedButton, setClickedButton] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleClick = useCallback(
+    async (sortedGoods: () => Promise<Good[]>, buttonName: string) => {
+      if (clickedButton === buttonName) {
+        return;
+      }
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      setIsLoading(true);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      try {
+        setGoods(await sortedGoods());
+        setClickedButton(buttonName);
+        setIsLoading(false);
+      } catch {
+        setHasError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    }, [clickedButton],
+  );
 
-    <GoodsList goods={[]} />
-  </div>
-);
+  return (
+    <div className="App page page__container">
+      <h1 className="page page__title">
+        Dynamic list of Goods
+      </h1>
+
+      <div className="buttons">
+        <Button
+          dataCy="all-button"
+          onClick={() => handleClick(getAll, 'all-button')}
+          clickedButton={clickedButton}
+        >
+          Load all goods
+        </Button>
+
+        <Button
+          dataCy="first-five-button"
+          onClick={() => handleClick(get5First, 'first-five-button')}
+          clickedButton={clickedButton}
+        >
+          Load 5 first goods
+        </Button>
+
+        <Button
+          dataCy="red-button"
+          onClick={() => handleClick(getRedGoods, 'red-button')}
+          clickedButton={clickedButton}
+        >
+          Load red goods
+        </Button>
+      </div>
+
+      {isLoading
+        ? (
+          <div>
```

Comment:
do you need this `div`?

Reply:
> 

yeah, to place it like a divided element)